### PR TITLE
fix: adds enum validations to healtcheckprotocol field

### LINKED ISSUE #4182: :bug: Users are able to submit bad data to healthCheckProtocol
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

**What did you expect to happen:**

I created an AWSCluster Object 

```yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
kind: AWSCluster
metadata:
  finalizers:
  - awscluster.infrastructure.cluster.x-k8s.io
  labels:
    cluster.x-k8s.io/cluster-name: faiq-test
  name: faiq-test
  namespace: default
  ownerReferences:
  - apiVersion: cluster.x-k8s.io/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Cluster
    name: faiq-test
spec:
  bastion:
    allowedCIDRBlocks:
    - 0.0.0.0/0
    enabled: false
  controlPlaneEndpoint:
    host: <redacted>
    port: 6443
  controlPlaneLoadBalancer:
    crossZoneLoadBalancing: false
    healthCheckProtocol: HTTPS:6443/healthz
    scheme: internet-facing
  identityRef:
```

specifying `spec.controlPlaneLoadBalancer.healthCheckProtocol: HTTPS:6443/healthz`.

I expect this to fail in the validating webhooks because only [allow specifying the protocol](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/5538c89bdfe245afa76bd9b72c0f64c92a865325/pkg/cloud/services/elb/loadbalancer.go#L943)

The fix here would be in the validating webhook. 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by faiq**
**What type of PR is this?**
/kind bug 

**What this PR does / why we need it**:
Validates data submitted to healthCheck protocol field 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4182 

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Adds validation for healthCheckProtocol Field 
```

### CODE REVIEW

File: api/v1beta1/awscluster_types.go [hunk 1]
```diff
@@ -186,6 +186,7 @@ type AWSLoadBalancerSpec struct {
 
 	// HealthCheckProtocol sets the protocol type for classic ELB health check target
 	// default value is ClassicELBProtocolSSL
+	// +kubebuilder:validation:Enum=TCP;SSL;HTTP;HTTPS
```
Comment (Ankitasw) [inline api/v1beta1/awscluster_types.go:189]: We generally dont backport changes to previous api. So changes for v1beta2 would suffice. You could revert this change.

Comment (faiq) [reply api/v1beta1/awscluster_types.go:189]: sounds good. thank you

Comment (faiq) [reply api/v1beta1/awscluster_types.go]: I have removed it. Thank you.

**Pull request merged**